tests/Event/ReadModel/History/EventPlaceHistoryProjectorTest.php

```diff
@@ -0,0 +1,341 @@
+            new LocationId($newPlaceId->toString()),
+            new Calendar(CalendarType::permanent()),
+            new Theme('1.8.1.0.0', 'Rock')
+        );
+
+        $this->projector->handle(
+            (new DomainMessageBuilder())->create($eventCreated)
+        );
+    }
+
+    /** @test */
+    public function apply_event_copied(): void
+    {
+        $oldEventId = $this->uuid4();
+        $eventId = $this->uuid4();
+        $oldPlaceId = $this->uuid4();
+
+        $this->eventRepository
+            ->expects($this->once())
+            ->method('fetch')
+            ->with($oldEventId->toString())
+            ->willReturn($this->createMockDocument($oldPlaceId->toString()));
+
+        $this->repository
+            ->expects($this->once())
+            ->method('storeEventLocationStartingPoint')
+            ->with(
+                $eventId,
+                $oldPlaceId
+            );
+
+        $eventCopied = new EventCopied(
+            $eventId->toString(),
+            $oldEventId->toString(),
+            new Calendar(CalendarType::permanent())
+        );
+
+        $this->projector->handle(
+            (new DomainMessageBuilder())->create($eventCopied)
+        );
+    }
+
+    /** @test */
+    public function apply_location_updated_logs_error_when_document_does_not_exist(): void
+    {
+        $eventId = $this->uuid4();
+        $newPlaceId = $this->uuid4();
+
+        $this->eventRepository
+            ->expects($this->once())
+            ->method('fetch')
+            ->with($eventId->toString())
+            ->willThrowException(new DocumentDoesNotExist('Document not found'));
+
+        $this->logger
+            ->expects($this->once())
+            ->method('error')
+            ->with($this->stringContains('Failed to store location updated: Document not found'));
+
+        $locationUpdated = new LocationUpdated($eventId->toString(), new LocationId($newPlaceId->toString()));
+        $this->projector->handle(
+            (new DomainMessageBuilder())->create($locationUpdated)
+        );
+    }
+
+    /** @test */
+    public function apply_event_updated_from_udb2(): void
+    {
+        $eventId = $this->uuid4();
+        $oldPlaceId = $this->uuid4();
+        $newPlaceId = new UUID('28d2900d-f784-4d04-8d66-5b93900c6f9c');
+
+        $this->eventRepository
+            ->expects($this->once())
+            ->method('fetch')
+            ->with($eventId->toString())
+            ->willReturn($this->createMockDocument($oldPlaceId->toString()));
+
+        $this->repository
+            ->expects($this->once())
+            ->method('storeEventLocationMove')
+            ->with(
+                $eventId,
+                $oldPlaceId,
+                $newPlaceId
+            );
+
+        $eventUpdatedFromUDB2 = new EventUpdatedFromUDB2(
+            $eventId->toString(),
+            SampleFiles::read(__DIR__ . '/../../samples/event_with_existing_location.cdbxml.xml'),
```

Comment:
I think you should also write a test where the location is a dummy location. There is a sample file for this `tests/Event/samples/event_with_dummy_location.cdbxml.xml`

Reply:
Added this, with using `cdbIdExtractor::getRelatedPlaceCdbId()`